### DISCUSSION TITLE
New/Changed CSS Selectors in 11.11.0

### DISCUSSION THREAD

**Discussion by mica-alex** [palexdev/MaterialFX#43] [Archived]
Greetings,

Are you able to let me know or clarify what CSS selectors I should use to style the following items?

1. MFXPasswordField Hide/Show Icon (I need to change color)
<img width="257" alt="PasswordFieldIcon" src="https://user-images.githubusercontent.com/83238954/121182348-fe938c00-c830-11eb-9064-fa007c62c940.PNG">

2. MFXComboBox Text (I need to change color)
<img width="248" alt="ComboBoxText" src="https://user-images.githubusercontent.com/83238954/121182488-21be3b80-c831-11eb-9ef9-a063c5403280.PNG">

The combo box text used to get changed to white via some other CSS selector I have, but it no longer changes in this version.

**Comment by palexdev** [comment]
The icon is wrapped in a MFXIconWrapper, so the selectors should be:
`.mfx-password-field .mfx-icon-wrapper .mfx-font-icon`

However, I just did a quick test in the demo, but it doesn't change color, maybe I found the issue, I'll fit it in the next minor release

The text seems to work only if you force your stylesheet to override the control's style by loading the CSS file using getStylesheet().add(CSS File)

These bugs are annoying, I know I know, but it's not completely my fault, JavaFX handles CSS badly

About styles and CSS, I was thinking of implementing like a Theme API or something that handles my control's style, I don't have any idea about the implementation though

**Reply by mica-alex** [reply]
No problem. Bugs can be annoying, but they happen.

The theme API sounds like a great addition. Something on a smaller scale of that might also be just a light/dark mode that you can switch between easily (from code).

I was able to get the ComboBox coloring back to normal by adding `-mfx-text-fill: white;` to my CSS as well as `-fx-text-fill: white;`

**Reply by palexdev** [reply]
Strange because neither MFXTextField nor MFXPassowrdField have the `-mfx-text-fill` property

**Reply by mica-alex** [reply]
Interesting, it worked with -fx-text-fill up until v11.11.0. As long as it works though, all is well :)

Did you want me to file an issue for the password field icon CSS color not working?

**Reply by palexdev** [reply]
Well said haha :)

No, don't worry, I just figured out the issue. It's the same issue that also affects text fields and combo boxes lines when invalid
I'm fixing and testing it right now, I'll publish a new minor version soon

**Comment by palexdev** [comment]
Version 11.13.0 not only brings refreshed controls and APIs for MaterialFX but also a new improved Wiki for the project
Now components in the wiki also list their CSS selectors: [Wiki](https://github.com/palexdev/materialfx/wiki)

**Reply by aaronfisher-code** [reply]
Been trying to read through the new documentation for the CSS selectors for the MFXDatepicker in order to adjust the remaining purple elements to blue (The combobox labels/dateCell highlighting)
Would you happen to know the appropriate tags for this change? Or potentially know why these elements arent having their styles updated?





![image](https://user-images.githubusercontent.com/79825913/150704345-f5046933-99ba-47c3-b29a-155d304fb8d7.png)

**Reply by palexdev** [reply]
Yes, of course, like I said the CSS selectors are all listed in the Wiki
Also, you can take a look at this [Pickers.css](https://github.com/palexdev/MaterialFX/blob/main/demo/src/main/resources/io/github/palexdev/materialfx/demo/css/Pickers.css)
